### ISSUE TITLE
"NULL.0" when TADO is off

### ISSUE THREAD

**Issue by FlashGordon7** [nervetattoo/simple-thermostat#28]
I have a TADO and when it is off, it does not report the set temperature.
The attribute "temperature" is then "null"

![image](https://user-images.githubusercontent.com/13345880/53277105-9d4c5b00-3702-11e9-9d70-05bb28f96b7c.png)
 I have fixed it by changing to this(from line 90):
```javascript
      const {
        attributes: {
          min_temp: minTemp = null,
          operation_mode: mode,
          operation_list: modes = [],
          temperature: _temperature,
        },
      } = entity
      
      if (typeof _temperature != "number") {
        this._temperature = minTemp;
      } else {
        this._temperature = _temperature
      }
```

**Comment by stickystyle** [comment]
I'm seeing the same behaviour with my Nest thermostat when it goes into 'Eco' mode.

**Comment by hohenstaufen** [comment]
Same behaviour with KNX thermostat.

**Comment by MatteGary** [comment]
same behavior with daikin when the device was set to "only fan" mode.

**Comment by MatteGary** [comment]
@FlashGordon7 can you please explain where to apply your fix? i don't find the line of code you suggest

**Comment by nervetattoo** [comment]
Is this really a duplicate of #4?
Does the thermostat use a single temp control when not in `Eco` mode and then switches to range based when in `eco`?

**Comment by MatteGary** [comment]
@nervetattoo wouldn't it be easier just to do something like "not showing anything" in case target temp is null? every climate device has his own different behavior, I think it's easier to just don't show NULL at least

**Comment by anthonym21** [comment]
the code is all obsfucated now can't even fix it

**Comment by nervetattoo** [comment]
Yes @MatteGary, there needs to be better handling. It could fall back to `N/A`, but that and all other solutions that doesn't show the target temperature will be bad and cause confusion. I mean, if the entire temperature control disappears people will start reporting bugs on that as well.

What would help a lot is to see some examples of what the entity attributes look like when this happens. You can get that from the `/dev-state` page in HA. As you mention different devices/platforms work in different ways and without having the devices myself its hard to know what potential attribute combinations can exist.

@anthonym21 The actual source is in `index.js` but you need to run `yarn build` to produce the actual card file. Its strictly not obfuscated but transpiled for compatibility and minimized for size. It is still the same JS just a lot harder to read and modify :D

**Comment by MatteGary** [comment]
@nervetattoo here's my entity attributes.
I have a Daikin Climate unit

{
  "current_temperature": 21,
  "min_temp": 7,
  "max_temp": 35,
  "temperature": null,
  "target_temp_step": 1,
  "fan_mode": "Auto",
  "fan_list": [
    "1",
    "2",
    "3",
    "4",
    "5",
    "Auto",
    "Silence"
  ],
  "operation_mode": "off",
  "operation_list": [
    "fan_only",
    "dry",
    "cool",
    "heat",
    "auto",
    "off"
  ],
  "swing_mode": "Off",
  "swing_list": [
    "3D",
    "Horizontal",
    "Off",
    "Vertical"
  ],
  "friendly_name": "Camera",
  "supported_features": 705,
  "icon": "mdi:hotel"
}

**Comment by hohenstaufen** [comment]
@nervetattoo here is the entity attributes for a KNX thermostat, maybe it can be useful.
```
{
  "current_temperature": 22.2,
  "min_temp": null,
  "max_temp": null,
  "temperature": null,
  "target_temp_step": 0.5,
  "operation_mode": "Comfort",
  "operation_list": [
    "Auto",
    "Comfort",
    "Standby",
    "Night",
    "Frost Protection"
  ],
  "friendly_name": "Room",
  "supported_features": 129
}
```

**Comment by nervetattoo** [comment]
> @nervetattoo here is the entity attributes for a KNX thermostat, maybe it can be useful.

Do you mind showing what it looks like when it is in other modes? I think that is the source of some issues

**Comment by jamesw4** [comment]
I am getting same issue with nest when its in eco mode. When in eco mode the temperature value of the climate entity is null. In all other modes it shows what the temperature is set to.

**Comment by dheath73** [comment]
I'm also getting same issue with nest when its in eco mode. When eco mode is set the temperature value of the climate entity is 'null'.

**Comment by crzykidd** [comment]
I see the same issue on ecobee 4 when the thermostat is in mode "off"

**Comment by jamesw4** [comment]
This logic seems broken, N/A will never be returned because of the statements that follow...........

function formatNumber(number, decimals = 1) {
  const [int, dec] = String(number).split('.')
  if (Number.isNaN(int)) {
    return 'N/A'
  }
  if (decimals) {
    return `${int}.${dec || '0'}`
  } else {
    return Math.round(int)
  }
}

**Comment by jamesw4** [comment]
In addition to my last comment I'm finding Number.isNaN to not work how I would expect. I've changed my copy slightly by just checking to start with if its null, if so display N/A otherwise carry out the other logic for decimals.

With the latest version I have replaced the section of code

function ct(t,e=1){const[s,i]=String(t).split(".");return Number.isNaN(s)?"N/A":e?`${s}.${i||"0"}`:Math.round(s)}

with

function ct(t,e=1){const[s,i]=String(t).split(".");return null==t?"N/A":e?`${s}.${i||"0"}`:Math.round(s)}

**Comment by dustins1983** [comment]
I absolutely love this card, but I cant use it until this is fixed. Not very wife friendly. PLEASE fix or show us how to!! It used to display set temp, even when off.

**Comment by nervetattoo** [comment]
This should be fixed in 0.22. Please reopen the issue if the problem persists
